### DISCUSSION TITLE
Vue components not re-rendered after changes

### DISCUSSION THREAD

**Discussion by skyroses** [livewire/livewire#1949] [1. Help] [answered]
### Exact steps to reproduce:

To see the problem, you can go [here](http://livewire-vue-test.herokuapp.com) and select at least one webmaster. These are components with user search. After some change occurs in the form, the second component does not start (managers).
Repo: https://github.com/moonofmylife/livewire-vue-test

1. composer and npm package install + npm run dev
2. php artisan migrate
3. php artisan db:seed
4. php artisan serve and visit site
5. select user in "webmaster" field

### Stripped-down, copy-pastable code snippets

**Context**
- Livewire version: 2.2.9
- Laravel version: 7.27.0
- Browser: Chrome

**Comment by joshhanley** [comment]
@moonofmylife you will probably need to make sure you have a wire:ignore on your Vue component, otherwise Livewire will probably cause it issues.

Can you try that to see if that works?

@danharrin can you move this to discussions while we diagnose what the issue is?

**Comment by skyroses** [comment]
@joshhanley yes, I tried it. This does not lead to anything good

**Comment by joshhanley** [comment]
@moonofmylife what happens when you add it?

The reason I ask is because, the way Livewire works, it breaks JavaScript libraries when Livewire changes DOM elements.

To stop that, you need to put wire:ignore on an element that is used by any other JavaScript, such as Vue. 

But that does lead to an issue where Livewire doesn’t know about anything inside of the element, as it essentially becomes a black box. The only way to communicate in and out is through events. 

Can I ask why you need Vue when Livewire is capable of all of this on its own?

**Comment by skyroses** [comment]
@joshhanley it's useless. After any change in the form, the Vue component is not rendered again. 
I use the Vue component because it fully meets all my requirements. Inside, a very functional multiselect component is used. I didn't find any analogs with similar functionality for Livewire. 
The livewire-vue package promises full-fledged Vue work together with Livewire on the same page, but it doesn't work properly and for some reason developers just ignore a bunch of issues in livewire-vue repo.

**Reply by joshhanley** [reply]
Ah ok fair enough. Yeah I don't have any experience using the adapter itself.

I was under the impression that it enabled Livewire and VueJS to exist on the same page (as they both hijack the DOM) but if it allows them to work together then that's even better!

I saw your below comment, so did you get it all working ok?

**Reply by skyroses** [reply]
@joshhanley unfortunately, this method only works if the element is not initially hidden or subsequently hidden. I have to instead of hiding or appearing with IF conditions, hide the element via css property (add class d-none in case of bootstrap) - this is not very convenient

**Comment by skyroses** [comment, marked as answer]
I noticed this strange thing - if the div element where the Vue component should be rendered is given a static id, then everything works fine
